source/fem-dem/vans_assemblers.cc

```diff
@@ -2256,13 +2315,23 @@ VANSAssemblerFPIProjection<dim>::assemble_rhs(
         -scratch_data.particle_two_way_coupling_force_values[q] / density;
       const Tensor<1, dim> &fluid_drag =
         -scratch_data.particle_drag_values[q] / density;
+      const double &beta_drag =
+        scratch_data.particle_momentum_transfer_coefficient_values[q];
+      const Tensor<1, dim> particles_velocity =
+        scratch_data.particle_velocity_values[q];
+      const Tensor<1, dim> velocity = scratch_data.velocity_values[q];
 
       // Store JxW in local variable for faster access;
       const double JxW = JxW_vec[q];
 
       // Calculate the strong residual for GLS stabilization
       // Drag Force and other two-way coupling forces
-      strong_residual[q] -= (fluid_drag + two_way_coupling_force);
+      strong_residual[q] -=
+        ( // This term is zero if the coupling is not explicit
+          fluid_drag
+          // This term is zero if the coupling is explicit
+          - beta_drag * (velocity - particles_velocity) +
+          two_way_coupling_force);
```

Comment:
I would reframe the comments to be a bit clearer here. I would remove those from within the equation and instead explain it thoroughly before the equation blocks. that will make the outcome clearer in my opinion.